enhancements/builds/volume-mounted-resources.md

```diff
@@ -0,0 +1,297 @@
+Those will be discussed in the future enhancement at that time.
+
+### Implementation Details/Notes/Constraints [optional]
+
+We will need to introduce a new mechanism in the build api which allows the user to indicate that they want to inject "volume" content into the build.
+Initially the only allowed volume types will be ConfigMaps, Secrets, and ephemeral CSI volumes.
+The api will otherwise be similar to the existing secret/configmap injection api in which users identify the configmap/secret and the target path for injection.
+
+This will be done by adding our own `Volume[]` field to the Source and Docker strategy structs.
+The `Volume[]` field will allow defining volumes to be mounted to the build pod in the same way that a normal pod allows for this.
+However, the types of volumes that can be defined will be restricted to those explictly supported by Builds.
+Similarly a `VolumeMount[]` field will be added, but without the MountPropagation and SubPath fields.
+MountPropagation and SubPath can be considered for support in the future.
+
+These fields will be wired, via the build controller, directly to the build pod that is constructed.
+In addition all mounts into the pod will be done at a path of our choosing, not the VolumeMount path specified, to ensure the user cannot overwrite critical function inside the build pod and use it as an escalation pathway.
+Example - all volumes are mounted under `/var/run/openshift.io/volumes` in the build pod containers.
+
+The logic that invokes buildah will then pass the mounted directories as transient volume mount arguments.
+The mount path provided to buildah will be determined from the VolumeMount specification.
+Arguments such as "read only" will also be inferred.
+
+Builds today have default mounts that are provided via other means:
+
+1. Entitlement data from the node, mounted at `/run/secrets/etc-pki-entitlement`, `/run/secrets/redhat.repo`, and `run/secrets/rhsm` inside the build pod.
+2. The cluster trust bundle - this is optionally mounted at `/etc/pki/ca-trust` via the BuildConfig spec's `mountTrustedCA` option.
+
+If a build volume's mount path conflicts with any of the paths specified above, the build volume mount will override the default.
+
+Proposed api/structs:
+Note:  DockerBuildStrategy will be updated in the same way.
+
+```go
+// SourceBuildStrategy defines input parameters specific to an Source build.
+type SourceBuildStrategy struct {
+  // From is reference to an DockerImage, ImageStream, ImageStreamTag, or ImageStreamImage from which
+  // the docker image should be pulled
+  From kapi.ObjectReference
+
+  // PullSecret is the name of a Secret that would be used for setting up
+  // the authentication for pulling the Docker images from the private Docker
+  // registries
+  PullSecret *kapi.LocalObjectReference
```

Comment:
Same comment about reusing the object reference types.

Reply:
Existing API in builds